blocks/embedded-form/embedded-form.js

```diff
@@ -0,0 +1,152 @@
+  const card = div({ class: 'card' });
+  card.append(...cardWrapper.children);
+  cardWrapper.append(card);
+  // remove button containers
+
+  card.querySelectorAll('.button-container').forEach((container) => {
+    [...container.children].forEach((node) => {
+      node.classList.remove('button');
+      node.classList.remove('primary');
+    });
+    container.classList.remove('button-container');
+  });
+}
+
+function validateInput(input) {
+  const formEl = input.closest('.Vlt-form__element');
+  const isFocused = document.activeElement === input;
+  const isValid = input.validity.valid;
+  const isDirty = formEl.classList.contains('Vlt-form__element--dirty');
+
+  if (isFocused) {
+    // remove all alerts when focused
+    formEl.classList.remove('Vlt-form__element--error');
+    formEl.classList.remove('Vlt-form__element--valid');
+    return;
+  }
+
+  if (isValid) {
+    formEl.classList.remove('Vlt-form__element--error');
+  } else {
+    formEl.classList.add('Vlt-form__element--error');
+  }
+
+  if (isValid && isDirty) {
+    formEl.classList.add('Vlt-form__element--valid');
+  } else {
+    formEl.classList.remove('Vlt-form__element--valid');
+  }
+}
+
+function submitForm() {
+  const form = document.querySelector("[data-form-type='lead form: apps: contact sales: in page']");
+  form.querySelectorAll('input[required], select[required]').forEach((input) => {
+    validateInput(input);
+  });
+  // there's a captcha that needs to be integrated. Thus, there will always be one error flag.
+  if (form.querySelectorAll('.Vlt-form__element--error').length <= 1) {
+    // fill in composite form fields
+    setFormValue('phonenumber', document.getElementById('dialing-code').value + document.getElementById('phone-number-local').value);
+    const url = form.getAttribute('action');
+
+    // submit form here.
+    const formData = new FormData();
+    form.querySelectorAll('.Vlt-form__element input, .Vlt-form__element select, input[type="hidden"]').forEach((input) => {
+      if (input.name !== '') {
+        formData.append(input.name, input.value);
+      }
+    });
+    fetch(url, {
+      method: 'post',
+      body: formData,
+    })
+      .then((response) => {
+        document.querySelector('.quote-form .right-column .form').classList.add('submitted');
+        if (response.redirected && response.url.includes('success')) {
+          document.querySelector('.quote-form .right-column .thank-you').classList.add('success');
+        } else {
+          // Content to show if form submission wasn't successful?
+        }
+      });
+  }
+}
+
+async function decorateRightColumn(formWrapper) {
+  const rightColumn = div({ class: 'right-column' });
+
+  const htmlFile = formWrapper.querySelector('p').textContent;
+  formWrapper.querySelector('p').remove();
+  formWrapper.querySelector('hr').remove();
+
+  const thankYouMesage = [...formWrapper.children];
+  rightColumn.append(div({ class: 'thank-you' }, ...thankYouMesage));
```

Comment:
copy paste for a simple failure message. something like `There was an error trying to submit the form. Please try again later.`
Ideally in the failure case the form is not hidden, so the user can try again.

Reply:
Done!